File: Java/Remove Duplicates from Sorted List.java
// 1ms, 9.60%; 41.1MB, 7.14%
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode tmp = head;
        while (tmp != null) {
            ListNode _next = tmp.next;
            while (_next != null && _next.val == tmp.val) {
                _next = _next.next;
            }
            tmp.next = _next;
            tmp = _next;
        }
        return head;
    }
}

File: Go/Remove Duplicates from Sorted List.go
// 4ms, 86.97%; 3.1MB, 100.00%
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
	cur := head
	for cur != nil {
		cur.Next = nextDifferent(cur, cur.Val)
		cur = cur.Next
	}
	return head
}

func nextDifferent(head *ListNode, val int) *ListNode {
	node := head
	for node != nil && node.Val == val {
		node = node.Next
	}
	return node
}
